src/datumaro/components/annotation.py

```diff
@@ -842,6 +843,21 @@ def __eq__(self, other):
         inter_area = self_polygon.intersection(other_polygon).area
         return abs(self_polygon.area - inter_area) < CHECK_POLYGON_EQ_EPSILONE
 
+    def _get_shoelace_area(self):
```

Comment:
@sooahleex 
If you are still interested in Rust, porting this Python function into Rust would be a good starting point.